src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -92,6 +92,8 @@ def convert_sf_to_sp_type(
     if column_type_name == "BINARY":
         return BinaryType()
     if column_type_name == "TEXT":
+        if internal_size > 0 and internal_size <= StringType._MAX_LENGTH:
```

Comment:
I tested it out on server side - `Error: Invalid character length: 0. Must be between 1 and 16,777,216. (line 4)`

Reply:
I wonder if we should fail if the size is not within the expected range when initializing `StringType` 🤔

Reply:
I'm following the logic followed by `DecimalType` a few lines below, which auto adjusts the input `precision` and `scale` to sensible ranges.

Right now, this function is called by values we get from result metadata. I'm not sure how reliable it is but if length value = 0 (which is possible in case when string length is not explicitly defined), then we should treat it as `StringType()` instead of raising error. 

I could possible raise error when length is negative value or great than `MAX_LENGTH`.

Reply:
Yeah negative value makes sense. I'm not particularly sure about `MAX_LENGTH` because this might change and is not controlled by client.